### PR TITLE
DOC: complete parameter server implementation

### PR DESCRIPTION
**Pull request by stsievert**
**What does this PR implement?**
It completes the implementation of the parameter server in the documentation. Right now, it has invalid syntax: https://docs.dask.org/en/latest/futures.html#example-parameter-server.

**Reference issues/PRs**
I saw this issue when linked from https://github.com/dask/dask-examples/pull/158#discussion_r459106996.

- [n/a] Tests added / passed
- [n/a] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: docs/source/futures.rst [hunk 1]
```diff
@@ -799,25 +809,30 @@ Example: Parameter Server
...
-       params = ps.get('parameters')
+    ps_future = client.submit(ParameterServer, actor=True)
+    ps = ps_future.result()
 
-   for batch in batches:
+    ps.put('parameters', np.random.random(1000))
+    for k in range(20):
+       params = ps.get('parameters').result()
+       new_params = train(params)
+       ps.put('parameters', new_params)
```
Comment (mrocklin) [inline docs/source/futures.rst:833]: The indentation here seems slightly off

**Comment by mrocklin** [comment]
Thanks for doing this @stsievert !

If you ever wanted to do something more real as a dask example that might also be fun :+1:

**Comment by stsievert** [comment]
> If you ever wanted to do something more real as a dask example that might also be fun 👍

That's part of why I submitted this PR. I am planning for a PR to Dask-ML to provide training for PyTorch models.

How experimental are Actors? Could they go in Dask-ML for the distributed training implementation?

**Comment by mrocklin** [comment]
> How experimental are Actors?

Not so experimental that anyone else is likely to change semantics.  But experimental enough that we can change things if we need to for your work :)

Really though, they're pretty simple.  I have a hard time thinking of parts of the current API that could break.

> Could they go in Dask-ML for the distributed training implementation?

I'm ok with it for situations where they're clearly the right fit.

**Comment by stsievert** [comment]
> Really though, they're pretty simple. 

That's good to hear. I have a use case where I should use them, so that provides more motivation. I've been holding off because there's no tools to monitor actors. I'm most interested in profiling, though it'd be nice to have a task stream too.

> I'm ok with it for situations where they're clearly the right fit.

They're not the right fit for this parameter server. The client should be the parameter server and task the gradients with computing the gradients. That's a lot simpler.

For the eventual Dask-ML PR, I have a rough implementation at https://github.com/stsievert/adadamp/pull/6. Better documentation is at https://docs.stsievert.com/adadamp.

**Comment by mrocklin** [comment]
> I'm most interested in profiling, 

I *think* that they should still participate in the statistical profiler (but I haven't checked, regardless it would be easy to do)

> though it'd be nice to have a task stream too

That might be a bit tougher.  One of the advantages of Actors is that there isn't any centralized reporting.  We could always set something up custom if necessary, but these probably won't drop into the existing dashboard system easily.

File: docs/source/futures.rst (hunk 1 above)
Comment (mrocklin) [reply docs/source/futures.rst:833]: Everything here is good to go except for this comment btw

Comment (stsievert) [reply docs/source/futures.rst]: Whoops, thanks.

**Comment by mrocklin** [comment]
Hrm, there are some interesting Parquet CI failures here.  They're almost certainly unrelated.  

@rjzamora @martindurant are these familiar to you?

**Comment by rjzamora** [comment]
> Hrm, there are some interesting Parquet CI failures here. They're almost certainly unrelated.
>@rjzamora @martindurant are these familiar to you?

Yup - #6346 *should* deal with them, but those tests need to be relaxed a bit to deal with pyarrow 1.0

**Comment by quasiben** [comment]
The rendering here doesn't look quite right:
Screen Shot 2020-07-28 at 9 45 46 AM

xref: https://docs.dask.org/en/latest/futures.html#example-parameter-server

**Comment by asmith26** [comment]
Thanks for providing this example @stsievert

I've been thinking about this example in the context of https://github.com/dask/dask-examples/pull/158, and I'm a bit confused about why Dask is needed.

In particular, I think the example is trying to demonstrate how you could create a very large model that couldn't fit on one machine using Dask to shared the (very large number of) parameters across a number of computers. However by running `params = ps.get('parameters').result()` you'll be pulling the (very large number of) parameters back to the Client/one machine, which could potentially crash that machine?

Sorry if I am mistaken. Thanks again!

**Comment by stsievert** [comment]
> I think the example is trying to demonstrate how you could create a very large model that couldn't fit on one machine

I mostly implemented this example to resolve invalid syntax (from a bad merge?). I suppose Dask might be used if the worker machines have GPUs and the client doesn't.

I think splitting the model up with two Actors would be a useful addition.
